pubsub/test/iam.test.js

```diff
@@ -14,159 +14,278 @@
+        assert(err);
+        assert(err.message = '"subscriptionName" is required!');
+        assert.equal(policy, undefined);
+      });
+    });
+    it('should handle error', function () {
+      var sample = getSample();
+      var error = 'error';
+      sample.mocks.subscription.iam.testPermissions.callsArgWith(1, new Error(error));
+      sample.program.testSubscriptionPermissions(subscriptionName, function (err, permissions) {
+        assert(err);
+        assert(err.message === 'error');
+        assert.equal(permissions, undefined);
+      });
+    });
+  });
+
+  describe('printUsage', function () {
+    it('should print usage', function () {
+      var program = getSample().program;
+
+      program.printUsage();
+
+      assert(console.log.calledWith('Usage: node iam RESOURCE COMMAND [ARGS...]'));
+      assert(console.log.calledWith('\nResources:\n'));
+      assert(console.log.calledWith('\ttopics'));
+      assert(console.log.calledWith('\tsubscriptions'));
+      assert(console.log.calledWith('\nCommands:\n'));
+      assert(console.log.calledWith('\tget NAME'));
+      assert(console.log.calledWith('\tset NAME'));
+      assert(console.log.calledWith('\ttest NAME'));
+      assert(console.log.calledWith('\nExamples:\n'));
+      assert(console.log.calledWith('\tnode iam topics get my-topic'));
+      assert(console.log.calledWith('\tnode iam topics set my-topic'));
+      assert(console.log.calledWith('\tnode iam topics test my-topic'));
+      assert(console.log.calledWith('\tnode iam subscriptions get my-subscription'));
+      assert(console.log.calledWith('\tnode iam subscriptions set my-subscription'));
+      assert(console.log.calledWith('\tnode iam subscriptions test my-subscription'));
+    });
+  });
+
+  describe('main', function () {
+    it('should call the right commands', function () {
+      var program = getSample().program;
+
+      sinon.stub(program, 'getTopicPolicy');
+      program.main(['topics', 'get']);
+      assert(program.getTopicPolicy.calledOnce);
+
+      sinon.stub(program, 'getSubscriptionPolicy');
+      program.main(['subscriptions', 'get']);
+      assert(program.getSubscriptionPolicy.calledOnce);
+
+      sinon.stub(program, 'setTopicPolicy');
+      program.main(['topics', 'set']);
+      assert(program.setTopicPolicy.calledOnce);
+
+      sinon.stub(program, 'setSubscriptionPolicy');
+      program.main(['subscriptions', 'set']);
+      assert(program.setSubscriptionPolicy.calledOnce);
+
+      sinon.stub(program, 'testTopicPermissions');
+      program.main(['topics', 'test']);
+      assert(program.testTopicPermissions.calledOnce);
+
+      sinon.stub(program, 'testSubscriptionPermissions');
+      program.main(['subscriptions', 'test']);
+      assert(program.testSubscriptionPermissions.calledOnce);
+
+      sinon.stub(program, 'printUsage');
+      program.main(['--help']);
+      assert(program.printUsage.calledOnce);
+
+      program.main(['topics', '--help']);
```

Comment:
Should we check that testTopicPermissions() wasn't called as well here, or is that overkill?